### ISSUE TITLE
Asymmetry in how EXTRA_C_FLAGS and EXTRA_CXX_FLAGS are handled

### ISSUE THREAD

**Issue by kennyweiss** [LLNL/blt#21]
There is an asymmetry in how we handle the ``EXTRA_C_FLAGS`` and ``EXTRA_CXX_FLAGS`` cache variables in ``SetupCompilerOptions.cmake``.  Here is the relevant part:

```bash
if(NOT CMAKE_CONFIGURATION_TYPES)
    ##########################################
    # Support Extra Flags for the C compiler.
    ##########################################
    if(EXTRA_C_FLAGS)
        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${EXTRA_C_FLAGS}")  # <-- adds to CMAKE_C_FLAGS
    endif()
    ...

    #############################################
    # Support Extra Flags for the C++ compiler.
    #############################################
    if(EXTRA_CXX_FLAGS)
        add_compile_options("${EXTRA_CXX_FLAGS}")    # <-- directly sets these as compile options
    endif()
    ...

endif()

```

We should be consistent in how we apply these or note the reason for the inconsistency.

We should probably also provide and ``EXTRA_FORTRAN_FLAGS`` variable.

**Comment by cyrush** [comment]
agreed, I also think these should be named "BLT_CXX_FLAGS", etc (EXTRA is a hold over before the BLT name was used)

**Comment by cyrush** [comment]
This was resolved in #22
